plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/cache/CachingHiveMetastore.java

```diff
@@ -119,31 +119,89 @@
     private final LoadingCache<String, Set<RoleGrant>> grantedPrincipalsCache;
     private final LoadingCache<String, Optional<String>> configValuesCache;
 
-    public static CachingHiveMetastore cachingHiveMetastore(HiveMetastore delegate, Executor executor, Duration cacheTtl, Optional<Duration> refreshInterval, long maximumSize, boolean partitionCacheEnabled)
+    public static CachingHiveMetastoreBuilder builder()
     {
-        return new CachingHiveMetastore(
-                delegate,
-                OptionalLong.of(cacheTtl.toMillis()),
-                refreshInterval
-                        .map(Duration::toMillis)
-                        .map(OptionalLong::of)
-                        .orElseGet(OptionalLong::empty),
-                Optional.of(executor),
-                maximumSize,
-                StatsRecording.ENABLED,
-                partitionCacheEnabled);
+        return new CachingHiveMetastoreBuilder();
     }
 
     public static CachingHiveMetastore memoizeMetastore(HiveMetastore delegate, long maximumSize)
     {
-        return new CachingHiveMetastore(
-                delegate,
-                OptionalLong.empty(),
-                OptionalLong.empty(),
-                Optional.empty(),
-                maximumSize,
-                StatsRecording.DISABLED,
-                true);
+        return builder()
+                .delegate(delegate)
+                .maximumSize(maximumSize)
+                .statsRecording(StatsRecording.DISABLED)
+                .build();
+    }
+
+    public static class CachingHiveMetastoreBuilder
+    {
+        private Optional<HiveMetastore> delegate = Optional.empty();
```

Comment:
`delegate` is not an optional field, it must eventually be set.
No point using Optional here.